### ISSUE TITLE
Javascript bundling

### ISSUE THREAD

**Issue by mehulkar** [11ty/eleventy#2490]
This may be a bit heretical / anti-11ty philosophy, but is there an opinion on how to compile and serve javascript? 

I want to add some dynamic portions to a website and so far have gotten away with inline `<script>` tags. But if I want to use a package (e.g. `web-vitals`) package, it's not clear how to do this. In one site I setup a separate `rollup` process, in another, I follow [this approach with a JS template](https://www.raresportan.com/eleventy-part-five/). Both are a bit convoluted, and I struggle to remember how it works when I come back to a project after some time.

**Describe the solution you'd like**
This may be solved with some documentation, but some kind of high level philosophy of how to take scripts, bundle & minify them, and put a reference in `<script src>` elements on the page would be useful. It would be even more useful if the script could optionally be inlined.

**Comment by mehulkar** [comment]
> if you didn't want to use the CDN version of web-vitals

correct. also not sure this is good practice. same origin is usually going to be better than cdn?

> `eleventyConfig.addPassthroughCopy({[webVitalsPath]: "scripts/web-vitals.min.js"});`

I guess this works, but I still need some glue code to _use_ the exported functions from this, so my code would have to import and use somehow. So some setup is still required. And in the cases of packages/modules that don't provide a single built file, you're still out of luck, trying to handpick files out and then trying to build them.

> Some docs on using Eleventy w/ Rollup or Webpack or Vite

@raresportan's approach in the blogpost works, but I wanted to check if there was anything better available!

> You could try making a PR

Ok, i'll try it out!

**Comment by Snapstromegon** [comment]
My take on this is to use rollup in combination with 11ty. I haven't made it into a plugin, since I'm still waiting for the official asset pipeline, but until then you can see my writeup of that solution here: https://www.hoeser.dev/blog/2021-02-28-11ty-and-rollup/

**Comment by Snapstromegon** [comment]
I finally tooke the time (partly thanks to this issue) and created [eleventy-plugin-rollup](https://www.npmjs.com/package/eleventy-plugin-rollup). This might be helpful for some.

**Comment by nhoizey** [comment]
I guess there's a way to do it with https://github.com/11ty/eleventy-plugin-bundle but I didn't find any docs/tutorial for this specific need, examples are provided only with Sass support for CSS.

**Comment by zachleat** [comment]
Among the other excellent shouts here I would also add the approaches we have documented here: https://www.11ty.dev/docs/assets/ and perhaps even more focused the esbuild example here https://www.zachleat.com/web/copy-paste/#bundling

Home base for a larger discussion here is at #3982
